frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1644,6 +1852,44 @@ mod tests {
 			assert!(MultiPhase::snapshot_metadata().is_none());
 			assert!(MultiPhase::desired_targets().is_none());
 			assert!(MultiPhase::queued_solution().is_none());
+			assert!(MultiPhase::signed_submissions().is_empty());
+		})
+	}
+
+	#[test]
+	fn early_termination_with_submissions() {
+		// an early termination in the signed phase, with no queued solution.
+		ExtBuilder::default().build_and_execute(|| {
+			// signed phase started at block 15 and will end at 25.
+			roll_to(14);
+			assert_eq!(MultiPhase::current_phase(), Phase::Off);
+
+			roll_to(15);
+			assert_eq!(multi_phase_events(), vec![Event::SignedPhaseStarted(1)]);
+			assert_eq!(MultiPhase::current_phase(), Phase::Signed);
+			assert_eq!(MultiPhase::round(), 1);
+
+			// fill the queue with signed submissions
+			for s in 0..SignedMaxSubmissions::get() {
+				let solution = RawSolution { score: [(5 + s).into(), 0, 0], ..Default::default() };
+				assert_ok!(MultiPhase::submit(
+					crate::mock::Origin::signed(99),
+					solution,
+					MultiPhase::signed_submissions().len() as u32
+				));
+			}
+
+			// an unexpected call to elect.
```

Comment:
Maybe we can consider finalizing the signed phase in such cases as well. After all, we have a a lot of solutions already, it is just that they are not verified. Essentially, if we are in signed phase and we receive `::elect`, we finalize the signed phase, and then do `do_elect`. This can be made into an issue in the `Npos` project.

Reply:
Does the call to `Self::finalize_signed_phase()` at the beginning of `do_elect` already do what you are suggesting?

Reply:
you're right -- even this test has `MultiPhase::elect().unwrap();`, which means election went fine.

Reply:
ahh right -- yeah, this should already happen.